Build failing due to typescript peer dependency version from vue-tsc

#53 exposed a current build error. See comment [here](https://github.com/gimbal-ghost/gimbal-ghost/pull/53#issuecomment-2689652023). Issue is confirmed to exist on main so must be caused by upstream package dependency change. Need to update typescript to v5 to resolve.

```
$ npm run lint && npm run typecheck && node scripts/build.mjs

> gimbal-ghost@1.3.1 lint
> eslint --max-warnings 0


> gimbal-ghost@1.3.1 typecheck
> vue-tsc --project packages/renderer/tsconfig.json --noEmit

/project/node_modules/vue-tsc/index.js:34
            throw err;
            ^

ReferenceError: sys is not defined
    at eval (eval at get (/project/node_modules/typescript/lib/tsc.js:9[87](https://github.com/gimbal-ghost/gimbal-ghost/actions/runs/13580663624/job/37966411419?pr=53#step:3:88)65:178), <anonymous>:1:1)
    at Object.get (/project/node_modules/typescript/lib/tsc.js:98765:178)
    at Object.proxyCreateProgram (/project/node_modules/@volar/typescript/lib/node/proxyCreateProgram.js:33:64)
    at /project/node_modules/typescript/lib/tsc.js:98765:107
    at Object.<anonymous> (/project/node_modules/typescript/lib/tsc.js:101378:3)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
    at Module.load (node:internal/modules/cjs/loader:1076:32)
    at Function.Module._load (node:internal/modules/cjs/loader:[91](https://github.com/gimbal-ghost/gimbal-ghost/actions/runs/13580663624/job/37966411419?pr=53#step:3:92)1:12)
    at Module.require (node:internal/modules/cjs/loader:1100:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

--- a/package.json
+++ b/package.json
@@ -55,7 +55,7 @@
         "stream-transform": "^3.0.4",
         "tailwind-scrollbar": "^1.3.1",
         "tailwindcss": "^3.0.24",
-        "typescript": "^4.6.3",
+        "typescript": "^5.7.3",
         "vite": "^4.5.2",
         "vue": "^3.4.21",
         "vue-tsc": "^2.0.7"

